plugins/hypervisors/vmware/src/main/java/com/cloud/storage/resource/VmwareStorageProcessor.java

```diff
@@ -1860,11 +1863,13 @@ public Answer createTemplateFromSnapshot(CopyCommand cmd) {
                 }
                 vmMo = clonedVm;
             }
+            s_logger.debug(String.format("Exporting volume to export path [%s], with VM config [%s].", exportPath, _gson.toJson(vmMo)));
             vmMo.exportVm(exportPath, exportName, false, false);
 
             return new Pair<>(diskDevice, disks);
         } finally {
             if (clonedVm != null) {
+                s_logger.debug(String.format("Detaching all VM [%s] disks and destroying it.", _gson.toJson(clonedVm)));
```

Comment:
detaching disks and destroy VM logged multiple times, can add these to a private method with log , detach disks + destroy VM ? or  add log to VirtualMachineMO, which covers all method calls.